matlab_octave/picard.m

```diff
@@ -133,18 +137,35 @@
             ls_tries = value;
         case 'whiten'
             whiten = value;
+        case 'pca'
+            whitening_mode = 'pca';
+            n_components = value;
+        case 'centering'
+            centering = value;
         case 'verbose'
             verbose = value;
         otherwise
             error(['Parameter ''' param ''' unknown'])
     end
 end
 
+if whiten == false && n_components ~= size(X, 1),
+    error('PCA works only if whiten=true')
+end
 
-% Whiten the signals if needed
+if n_components ~= rank(X, 1),
```

Comment:
The rank() calculation occasionally leads to wrong results under Linux 64bit. For this reason an alternative rank calculation is implemented in pop_runica(). Should we use this one? It's called getrank()

Reply:
If yes, can you please tell me, what's the best way to make getrank() available in picard.m

Reply:
You can copy and paste the function from pop_runica() in picard.m
``` 
function tmprank2 = getrank(tmpdata);
    % Function originally written in EEGLAB by Arnaud Delorme
    tmprank = rank(tmpdata);
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %Here: alternate computation of the rank by Sven Hoffman
    %tmprank = rank(tmpdata(:,1:min(3000, size(tmpdata,2)))); old code
    covarianceMatrix = cov(tmpdata', 1);
    [E, D] = eig (covarianceMatrix);
    rankTolerance = 1e-7;
    tmprank2=sum (diag (D) > rankTolerance);
    if tmprank ~= tmprank2
        fprintf('Warning: fixing rank computation inconsistency (%d vs %d) most likely because running under Linux 64-bit Matlab\n', tmprank, tmprank2);
        tmprank2 = max(tmprank, tmprank2);
    end     
```